ReactiveFeedback/Feedback.swift

```diff
@@ -40,20 +56,27 @@ public struct Feedback<State, Event> {
         }
     }
 
+    /// Creates a Feedback which will perform effects on for certain state filtered by the predicate.
+    /// Each new effect cancels a previous one
+    /// - parameters:
+    ///    - predicate: A closure which defines weather effect should be performed to particular value of the `State`
+    ///    - effects: A sequence of the Events over time that mutate the State
     public init<Effect: SignalProducerConvertible>(
         predicate: @escaping (State) -> Bool,
         effects: @escaping (State) -> Effect
     ) where Effect.Value == Event, Effect.Error == NoError {
         self.events = { scheduler, state in
-            return state
+            return state.filter(predicate)
                 .flatMap(.latest) { state -> SignalProducer<Event, NoError> in
-                    guard predicate(state) else { return .empty }
                     return effects(state).producer
                         .enqueue(on: scheduler)
             }
         }
     }
 
+    /// Creates Feedback which will perform effects on for each state changes, canceling previously performed one
```

Comment:
`Creates a Feedback which will perform effect for each state changes, canceling previously performed ones`